File: python/rna-transcription/dna.py
def to_rna(dna):
    dict = {"G":"C", "C":"G", "T":"A", "A":"U"}
    dna = list(dna)
    rna = []
    for c in dna:
        rna.append(dict[c])
    return "".join(rna)

File: haskell/rna-transcription/DNA.hs
module DNA (toRNA) where

import Data.Maybe

rnaMap :: [(Char, Char)]
rnaMap = [('G', 'C'), ('C', 'G'), ('T', 'A'), ('A', 'U')]

toRNA :: String->String
toRNA dna = [Data.Maybe.fromJust $ lookup s rnaMap | s <- dna]
